core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java

```diff
@@ -233,6 +222,27 @@ private static ResponseConverterFunction responseConverter(
         return responseConverter;
     }
 
+    private static Type getActualReturnType(Method method) {
+        final Function<Method, Class<?>> returnTypeGetter;
+        final Function<Method, Type> genericReturnTypeGetter;
+
+        if (KotlinUtil.isSuspendingFunction(method)) {
+            returnTypeGetter = KotlinUtil::kFunctionReturnType;
+            genericReturnTypeGetter = KotlinUtil::kFunctionGenericReturnType;
+        } else {
+            returnTypeGetter = Method::getReturnType;
+            genericReturnTypeGetter = Method::getGenericReturnType;
+        }
```

Comment:
I don't see any advantage of using `Function`. Could we assign `Class` and `Type`?
```java
final Class<?> returnType;
final Type genericReturnType;
if (KotlinUtil.isSuspendingFunction(method)) {
    returnType = KotlinUtil.kFunctionReturnType(method);
    genericReturnType = KotlinUtil.kFunctionGenericReturnType(method);
} else {
    returnType = method.getReturnType();
    genericReturnType = method.getGenericReturnType();
}
```

Reply:
Thanks, fixed! I don't understand why I have done it that way.. 😓